src/axom/inlet/LuaReader.cpp

```diff
@@ -64,114 +54,105 @@ bool LuaReader::parseString(const std::string& luaString)
-    }
-    else
-    {
-      lua_getfield(m_luaState, -1, token.c_str());
-    }
-    if(lua_isnil(m_luaState, -1))
-    {
-      // variable not found
-      return false;
-    }
-    atGlobalScope = false;
-  }
-
-  return true;
+bool LuaReader::getInt(const std::string& id, int& value)
+{
+  return getValue(id, value);
 }
 
 
-bool LuaReader::getBool(const std::string& id, bool& value)
+bool LuaReader::getString(const std::string& id, std::string& value)
 {
-  if (!findVariable(id))
-  {
-    return false;
-  }
-  value = (bool)lua_toboolean(m_luaState, -1);
-  return true;
+  return getValue(id, value);
 }
 
+bool LuaReader::getIntMap(const std::string& id, std::unordered_map<int, int>& values) {
+  return getMap(id, values, sol::type::number);
+}
 
-bool LuaReader::getDouble(const std::string& id, double& value)
-{
-  if (!findVariable(id))
-  {
-    return false;
-  }
-  value = (double)lua_tonumber(m_luaState, -1);
-  return true;
+bool LuaReader::getDoubleMap(const std::string& id, std::unordered_map<int, double>& values) {
+  return getMap(id, values, sol::type::number);
 }
 
+bool LuaReader::getBoolMap(const std::string& id, std::unordered_map<int, bool>& values) {
+  return getMap(id, values, sol::type::boolean);
+}
+ 
+bool LuaReader::getStringMap(const std::string& id, std::unordered_map<int, std::string>& values) {
+  return getMap(id, values, sol::type::string);
+}
 
-bool LuaReader::getInt(const std::string& id, int& value)
-{
-  if (!findVariable(id))
-  {
+template <typename T>
+bool LuaReader::getValue(const std::string& id, T& value) {
+  std::vector<std::string> tokens;
+  axom::utilities::string::split(tokens, id, SCOPE_DELIMITER);
+  if (tokens.size() == 1 && m_lua[id].valid()) {
+    value = m_lua[id];
+    return true;
+  }
+  
+  if (!m_lua[tokens[0]].valid()) {
     return false;
   }
-  value = (int)lua_tonumber(m_luaState, -1);
-  return true;
+  sol::table t = m_lua[tokens[0]];
+  for (size_t i = 1; i < tokens.size()-1; i++) {
+    if (t[tokens[i]].valid()) {
+      t = t[tokens[i]];
+    } else {
+      return false;
+    }
+  }
+  if (t[tokens.back()].valid()) {
+    value = t[tokens.back()];
+    return true;
+  }
+  
+  return false;
 }
 
+template <typename T>
+bool LuaReader::getMap(const std::string& id, std::unordered_map<int, T>& values, sol::type type) {
+  std::vector<std::string> tokens;
+  axom::utilities::string::split(tokens, id, SCOPE_DELIMITER);
 
-bool LuaReader::getString(const std::string& id, std::string& value)
-{
-  if (!findVariable(id))
-  {
+  if (tokens.empty() || !m_lua[tokens[0]].valid()) {
     return false;
   }
-  value = std::string(lua_tostring(m_luaState, -1));
+  sol::table t = m_lua[tokens[0]];
+  for (size_t i = 1; i < tokens.size(); i++) {
+    if (t[tokens[i]].valid()) {
+      t = t[tokens[i]];
+    } else {
+      return false;
+    }
+  }
+
+  auto it = t.cbegin();
+  while (it != t.cend()) {
+    if ((*it).first.get_type() == sol::type::number 
```

Comment:
Add a comment clarifying what this if is doing.. "get only numbered items in table" or something.